### ISSUE TITLE
Profile picture (avatar) not shown in app

### ISSUE THREAD

**Issue by MelBourbon** [stefan-niedermann/nextcloud-deck#664]
**Describe the bug**
Avatar (profile) picture is not shown when user is assigned to a card in the Deck Android App even if it's selected on server and correctly shown in Nextcloud App. Instead the standard picture is shown for all users.

**To reproduce**
Issue is reproducible with an account from [try.nextcloud.com](https://try.nextcloud.com):
- [x] Yes
- [ ] No

1. Go to any card and assign yourself as responsible
2. Click on the card or open a list with the card
3. See that the profile picture is not shown

**Expected behavior**
When users have selected and assigned a specific profile picture on the server this picture should be shown in the app similar like it is shown in the Nextcloud Android App.

**Screenshots**
Not available and needed to explain issue.

**Versions**
 - Nextcloud: 18.0.7
 - Nextcloud Deck: 1.0.5
 - Nextcloud Android: 3.13.0
 - Nextcloud Android Deck: 1.9.7

**Smartphone (please complete the following information):**
- Device: OnePlus 5T
- Android-Version: 10
- App-Store:
  - [x] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [ ] F-Droid
  - [ ] Huawei AppGallery

**Are you using LDAP?**
- [ ] Yes
- [x] No

**Comment by stefan-niedermann** [comment]
This is a strange one. I have a few questions @MelBourbon 

- Is your user avatar shown in the top right of the app, where you can manage your accounts?
- Do other users of your instance also see no avatar or is it visible for them?
- Can you create a test account and share the credentials with info@niedermann.it so we can try to reproduce the problem?

**Comment by MelBourbon** [comment]
Hi @stefan-niedermann 
- No, only the placeholder is shown in the top right of the app (see screenshot attached)
- No, other users also only see the placeholder picture (second screenshot attached was made by another user)
- Test account is created and credentials will be send by mail

Screenhot 1:
![Screenshot_20200828-113530](https://user-images.githubusercontent.com/30600002/91548058-1f839400-e925-11ea-8ce2-ef25bb28824c.jpg)

Screenhot 2:
![Screenshot_20200828-114059](https://user-images.githubusercontent.com/30600002/91548064-21e5ee00-e925-11ea-838f-9215e8730ebe.jpg)

**Comment by stefan-niedermann** [comment]
Thank you very much. The first impression is, that your Nextcloud instance is not in the root directory of your domain but under a subfolder `example.com/nextcloud` - this might be the cause. You checked, that the issue is reproducible with an account from `try.nextcloud.com` - are you sure this is valid? I will try to have a closer look soon, but i am currently a bit busy.

**Comment by MelBourbon** [comment]
Yes, issue seems to be related to the point that instance is not in the root directory. I now checked it again with another instance by a friend which is in the root directory and there it is working.

**Comment by stefan-niedermann** [comment]
@desperateCoder i need a way to get the absolute path to an instance. Can you check if there is a property in the `capabilities` response or so? Maybe we have to store it in the database and make it accessible. In this case we should perform a migration which puts `example.com` there until the next sync (which might or might not change it to `example.com/mycloud` then).

**Comment by jvlker** [comment]
> 
> 
> Thank you very much. The first impression is, that your Nextcloud instance is not in the root directory of your domain but under a subfolder `example.com/nextcloud` - this might be the cause. 

I can confirm that issue. I got only the default avatars and hosted my nextcloud exactly in the described subfolder (url/nextcloud)

**Comment by desperateCoder** [comment]
@MelBourbon any chances, your server just has died or something?

![grafik](https://user-images.githubusercontent.com/6911132/91638915-1bce3b00-ea13-11ea-9e4f-c1cb25dd59a1.png)

**Comment by desperateCoder** [comment]
@MelBourbon The domain can't even be resolved (DNS lookup fails), something is quite broken there... I can confirm it worked a few hours ago (you got some nice photographing skills - respect!). Now as I wanted to fix this issue, the site is gone... 

Please have a look what's wrong there and give some feedback as soon as it is back up.

**Comment by desperateCoder** [comment]
@MelBourbon ok it seems to be back, but holy cow, it tells me the following:

> apps directory not found! Please put the Nextcloud apps folder in the Nextcloud folder or the folder above. You can also configure the location in the config.php file.

**Comment by MelBourbon** [comment]
@desperateCoder Sorry. Was not aware that you will need my instance to test. Based on this issue and as I always wanted to do I switched to the main url only (without `/nextcloud`). With this my instance is not the right one for testing this issue. Sorry for the inconveniences.

**Comment by desperateCoder** [comment]
@MelBourbon no problem buddy! My primary concern was, that your server just broke and no one has a clue why :sweat_smile: 

@jvlker would you mind providing us a test-account on your instance? If you send the credentials to info@niedermann.it, I'll have a look how we can fix this, so you and other users don't need to move your instance around. Would be much appreciated, after @MelBourbon stole our bug :stuck_out_tongue_closed_eyes:!

**Comment by jvlker** [comment]
:D 
@desperateCoder yes, I can grant you access to my instance. You'll get an email soon.

**Comment by desperateCoder** [comment]
@jvlker thank you very much for the test account, I was able to run some tests on it already.

"Unfortunately" the URL to the avatars seems to get built correctly, we already **do** take the `/nextcloud` into account, but somehow the avatar still doesn't want to get displayed... 

Please keep the test account a few days more, so @stefan-niedermann can test these days, too. 

Again: thanks for your help, we'll keep you on track for any news!

**Comment by stefan-niedermann** [comment]
Okay, my tests so far: I can confirm that this `curl` works:

```
curl -X GET 'https://<username>:<password>@<domain>/nextcloud/index.php/avatar/<username>/64' -H 'Accept: */*' -H 'OCS-APIRequest: true' -H 'Content-Type: application/json;charset=utf-8' > ~/Schreibtisch/test.jpg
```

I can also confirm that the correct url is requested:

```
https://<domain>/nextcloud/index.php/avatar/<username>/64
```

So: The `/nextcloud` subfolder does not seem to be the root cause (even if it seems to be since you both have issues and both have a similar setup).

Further investigation...

**Comment by stefan-niedermann** [comment]
@MelBourbon and @jvlker do your user ids contain a space character?

**Comment by jvlker** [comment]
@stefan-niedermann no, its similar to my github name, only letters.

**Comment by markushub** [comment]
I also can't see avatars. My nextcloud is running on vhost inside a sub directory.
Its hard to use deck on android without see the assignment in a list overview.
I have to go inside details, see the default icon and have to push it to see the name. 

Is the problem based on nextcloud server side to provide a avatar or on android side?
I hope it will be fixed soon.
Thank you for your great work with deck app.

**Comment by stefan-niedermann** [comment]
@markushub The issue is most probably caused by the Deck Android App.
We haven't found the root cause yet, but suspect it to be related to the fact, that the cloud is located in a subfolder.

---

Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue.

**Comment by stefan-niedermann** [comment]
Good news, we believe that we have been able to solve this issue completely. If you want, you can safely try the solution, installing the [latest `APK` directly from here](https://github.com/stefan-niedermann/nextcloud-deck/suites/1700393111/artifacts/31998649) - it will be installed parallel to your default Deck Android app.

It will be released with the next version `1.13.5`.

Please don't hesitate to open a new issue if the described problems are still present at the linked `APK` or in the released version `1.13.5` or higher! Thank you for reporting! :slightly_smiling_face:
